proto/redpanda/api/dataplane/v1alpha3/ai_agent.proto

```diff
@@ -0,0 +1,684 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha3;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/api/resource.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "redpanda/api/auth/v1/authorization.proto";
+import "redpanda/api/dataplane/v1/pipeline.proto";
+
+// Defines the AI Agent resource.
+message AIAgent {
+  option (google.api.resource) = {
+    type: "redpanda.api.dataplane.v1alpha3/AIAgent"
+    singular: "ai_agent"
+    plural: "ai_agents"
+  };
+
+  // AI Agent ID.
+  string id = 1 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
+
+  // User-friendly AI agent name.
+  string display_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  // Optional AI agent description.
+  string description = 3 [(buf.validate.field).string.max_len = 256];
+
+  // System Prompt for the AI agent behavior.
+  string system_prompt = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 10,
+    (buf.validate.field).string.max_len = 16384,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "System instructions that define how the AI agent should behave and respond."}
+  ];
+
+  // LLM configuration settings for the AI agent.
+  Provider provider = 5 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  int32 dimensions = 6 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.field_behavior) = IMMUTABLE,
+    (buf.validate.field).required = true
+  ];
+  string model = 7 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.field_behavior) = IMMUTABLE,
+    (buf.validate.field).required = true
+  ];
+
+  // List of Redpanda MCP server IDs that this AI agent can use.
+  repeated string mcp_server_ids = 8;
```

Comment:
hm, i think we need a message type value for this, so we can e.g. include/exclude tools.

@weeco mentioned this, some frameworks allow plugging an mcp, but you can provide an exclusion, inclusion regex for tool names.

what about

message MCPServer {
  string id = 1;
  included_tools_pattern = 2; // Default to including all / .*
  excluded_tools_pattern = 3; // Default to empty / disabled
  // Anything else?
}

might as well do map for fieldmask tbh

Reply:
What I have seen was the filter on the consumer side (agent that can configure the include/exclude)